ui/src/components/Input.tsx

```diff
@@ -1,42 +1,37 @@
-import { ChangeEventHandler, ReactNode } from 'react';
+import { ReactNode } from 'react';
 import classNames from 'classnames';
+import { useFormContext } from 'react-hook-form';
+import { t } from 'i18next';
 
 interface Props {
+  name: string;
   type: string;
   placeholder: string;
-  name: string;
-  value?: string;
-  required?: boolean;
   autoComplete?: string;
   children?: ReactNode;
-  onChange?: ChangeEventHandler<HTMLInputElement>;
-  error?: string;
 }
 
-const Input = ({ type, name, placeholder, value, required, autoComplete, children, onChange, error }: Props) => {
-  const style = classNames('input', children && 'pr-11', error && 'border border-red-500');
+const Input = ({ name, type, placeholder, autoComplete, children }: Props) => {
+  const {
+    register,
+    formState: { errors },
+  } = useFormContext();
+  const style = classNames('input', children && 'pr-11', errors[name] && 'border border-red-500');
 
   return (
     <div className="relative w-full">
-      <input
-        className={style}
-        type={type}
-        name={name}
-        placeholder={placeholder}
-        value={value}
-        required={required}
-        onChange={onChange}
-        autoComplete={autoComplete}
-      />
+      <input className={style} type={type} placeholder={placeholder} autoComplete={autoComplete} {...register(name)} />
       {children}
-      <div className="text-red-500">{error}</div>
+      <div className="text-left text-red-500">
+        {errors?.[name] && <div>{t(errors[name]?.message?.toString() || 'validation.error')}</div>}
```

Comment:
Nie jestem przekonany co do tego zapisu (validation.error), lecz nie powoduje warningów. Jak inaczej to zrobić?

Reply:
tutaj nie powinniśmy wywoływać funkcji t z atrybutem message. Message jest już zwracany jako przetłumaczony string. 
```suggestion
        {errors?.[name] && <div>{errors[name]?.message?.toString() || t('validation.error')}</div>}
```

Reply:
można przygotować dodatkową funckje która będzie sprawdzać jakiego typu jest `errors[name]?.message`, ale ja nie widzę problemu żeby to zostawic w ten sposób. To że krzyczy że może być undefined to właściwie dobrze, przynajmniej wymusza na nas żeby zapewnić jakąś domyślną wiadomośc